patroni/ctl.py

```diff
@@ -235,6 +242,39 @@ def get_cursor(cluster, connect_parameters, role='master', member=None):
     return None
 
 
+def get_members(cluster, cluster_name, member_names, role, force, action):
+    candidates = {m.name: m for m in cluster.members}
+
+    if not force or role:
+        output_members(cluster, cluster_name)
+
+    if role:
+        role_names = [m.name for m in get_all_members(cluster, role)]
+        if member_names:
+            member_names = list(set(member_names) & set(role_names))
+        else:
+            member_names = role_names
+
+    if not member_names:
```

Comment:
If role_members or member_names had been set before being filtered out by lines 251-256, we should not ask the user again to pick up the member.